test/e2e/dra/dra.go

```diff
@@ -118,6 +118,29 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 				return nil
 			}, 20*time.Second, 200*time.Millisecond).Should(gomega.BeNil())
 		})
+		ginkgo.It("must unprepare resources for force-deleted pod", func() {
+			parameters := b.parameters()
+			claim := b.externalClaim(v1.AllocationModeImmediate)
+			pod := b.podExternal()
+			zero := int64(0)
+			pod.Spec.TerminationGracePeriodSeconds = &zero
+			pod.ObjectMeta.DeletionGracePeriodSeconds = &zero
```

Comment:
`DeletionGracePeriodSeconds` cannot be set when creating an object. I suppose the apiserver silently removes it.

Reply:
I just wanted to set all related attributes to trigger force deletion. Strictly speaking, specifying `, metav1.DeleteOptions{GracePeriodSeconds: &zero` should be enough, but I wanted to be absolutely sure I did everything possible.

I can remove this line if you want, no problem with that.

Reply:
Can you remove this?

We can then merge this test. It passes also for me.

However, it would be good to understand why kubelet now calls NodeUnprepareResource and whether it does that reliably.

Reply:
done.

Yes, I'm working on that. So far my impression is that Kubelet did the same before, but our setup has changed. As far as I remember node plugin pod was removed before the UnprepareResource call when we discovered that issue. It doesn't happen anymore, so Kubelet has time to garbage collect force-deleted pod and call UnprepareResource.

I'll investigate this further.